pkgs/desktops/gnome/extensions/update-extensions.py

```diff
@@ -82,39 +83,43 @@ def generate_extension_versions(
     """
 
     # Determine extension version per shell version
-    extension_versions: Dict[ShellVersion, ExtensionVersion] = {}
+    # type: { ShellVersion: { 'version': int, 'pk': int } }
```

Comment:
You can use https://docs.python.org/3.8/library/typing.html#typing.TypedDict

Reply:
I cannot say I like that type name, but at least it's working